File: frontend/src/utils/styleCategory.ts
type CategoryOption = { label: string; value: string };

export const CATEGORY_CODE_OPTIONS: CategoryOption[] = [
  { label: '女装', value: 'WOMAN' },
  { label: '男装', value: 'MAN' },
  { label: '童装', value: 'KIDS' },
  { label: '女童装', value: 'WCMAN' },
  { label: '男童装', value: 'MCMAN' },
  { label: '男女同款', value: 'UNISEX' },
  { label: '运动装', value: 'SPORT' },
  { label: '内衣', value: 'UNDERWEAR' },
  { label: 'T恤', value: 'T_SHIRT' },
  { label: '衬衫', value: 'SHIRT' },
  { label: '卫衣', value: 'HOODIE' },
  { label: '毛衣', value: 'SWEATER' },
  { label: '夹克', value: 'JACKET' },
  { label: '大衣', value: 'COAT' },
  { label: '风衣', value: 'TRENCH_COAT' },
  { label: '羽绒服', value: 'DOWN_JACKET' },
  { label: '棉服', value: 'PADDED_JACKET' },
  { label: '西装', value: 'SUIT' },
  { label: '马甲', value: 'VEST' },
  { label: '连衣裙', value: 'DRESS' },
  { label: '半身裙', value: 'JUPE' },
  { label: '短裤', value: 'SHORTS' },
  { label: '长裤', value: 'TROUSERS' },
  { label: '牛仔裤', value: 'JEANS' },
  { label: '休闲裤', value: 'CASUAL_PANTS' },
  { label: '运动裤', value: 'SWEATPANTS' },
  { label: '打底衫', value: 'BASE_SHIRT' },
  { label: '打底裤', value: 'BASE_PANTS' },
  { label: '瑜伽服', value: 'YOGA_WEAR' },
  { label: '防晒服', value: 'SUN_PROTECTION' },
  { label: '家居服', value: 'LOUNGEWEAR' },
  { label: '泳装', value: 'SWIMWEAR' },
  { label: '工作服', value: 'WORKWEAR' }
];

export const SEASON_CODE_OPTIONS: CategoryOption[] = [
  { label: '春季', value: 'SPRING' },
  { label: '夏季', value: 'SUMMER' },
  { label: '秋季', value: 'AUTUMN' },
  { label: '冬季', value: 'WINTER' },
  { label: '春夏', value: 'SPRING_SUMMER' },
  { label: '秋冬', value: 'AUTUMN_WINTER' },
];

/**
 * 品类英文 token → 中文 全量映射（key 统一为大写，大小写均可命中）。
 * 数据库里品类字段会混杂代码、大小写英文或中文，这里保证任何形式都能转成中文展示。
 */
const CATEGORY_CN_MAP: Record<string, string> = {
  // 主分类
  WOMAN: '女装',
  WOMEN: '女装',
  LADY: '女装',
  MAN: '男装',
  MEN: '男装',
  KID: '童装',
  KIDS: '童装',
  CHILD: '童装',
  CHILDREN: '童装',
  WCMAN: '女童装',
  MCMAN: '男童装',
  UNISEX: '男女同款',
  SPORT: '运动装',
  UNDERWEAR: '内衣',
  // 上装类
  TOP: '上衣',
  UPPER: '上衣',
  TOPS: '上衣',
  T_SHIRT: 'T恤',
  TSHIRT: 'T恤',
  SHIRT: '衬衫',
  BLOUSE: '衬衫',
  HOODIE: '卫衣',
  SWEATER: '毛衣',
  KNITWEAR: '针织衫',
  JACKET: '夹克',
  COAT: '大衣',
  TRENCH_COAT: '风衣',
  DOWN_JACKET: '羽绒服',
  PADDED_JACKET: '棉服',
  SUIT: '西装',
  VEST: '马甲',
  BASE_SHIRT: '打底衫',
  // 下装类
  SKIRT: '半身裙',
  JUPE: '半身裙',
  SHORTS: '短裤',
  SHORT: '短裤',
  TROUSERS: '长裤',
  PANTS: '裤子',
  JEANS: '牛仔裤',
  CASUAL_PANTS: '休闲裤',
  SWEATPANTS: '运动裤',
  BASE_PANTS: '打底裤',
  LEGGINGS: '打底裤',
  // 连衣裙 / 连体
  DRESS: '连衣裙',
  ONEPIECE: '连衣裙',
  JUMP_SUIT: '连体裤',
  OVERALLS: '连体裤',
  // 功能/场景类
  YOGA_WEAR: '瑜伽服',
  SUN_PROTECTION: '防晒服',
  LOUNGEWEAR: '家居服',
  SWIMWEAR: '泳装',
  SWIMSUIT: '泳装',
  WORKWEAR: '工作服',
  UNIFORM: '工作服',
  ACCESSORIES: '配饰',
  UNDECLARED: '未分类',
};

export const toCategoryCn = (value: unknown, options?: CategoryOption[]): string => {
  const raw = String(value ?? '').trim();
  if (!raw) return '-';
  const upper = raw.toUpperCase();
  const opt = Array.isArray(options)
    ? options.find((o) => String(o?.value ?? '').trim().toUpperCase() === upper)
    : undefined;
  if (opt?.label) return opt.label;
  return CATEGORY_CN_MAP[upper] || raw;
};

export const normalizeCategoryQuery = (value: unknown): string => {
  const raw = String(value ?? '').trim();
  if (!raw) return '';
  const upper = raw.toUpperCase();
  if (upper === 'WOMAN' || upper === 'WOMEN' || raw === '女装') return 'WOMAN';
  if (upper === 'MAN' || upper === 'MEN' || raw === '男装') return 'MAN';
  if (upper === 'KID' || upper === 'KIDS' || upper === 'CHILD' || upper === 'CHILDREN' || raw === '童装') return 'KIDS';
  if (upper === 'WCMAN' || raw === '女童装') return 'WCMAN';
  if (upper === 'UNISEX' || raw === '男女同款') return 'UNISEX';
  if (upper === 'SPORT' || raw === '运动装') return 'SPORT';
  if (upper === 'UNDERWEAR' || raw === '内衣') return 'UNDERWEAR';
  return raw;
};

export const toSeasonCn = (value: unknown): string => {
  const raw = String(value ?? '').trim();
  if (!raw) return '-';
  const upper = raw.toUpperCase();
  const map: Record<string, string> = {
    SPRING: '春季',
    SUMMER: '夏季',
    AUTUMN: '秋季',
    WINTER: '冬季',
    SPRING_SUMMER: '春夏',
    AUTUMN_WINTER: '秋冬',
  };
  return map[upper] || raw;
};

export const normalizeSeasonQuery = (value: unknown): string => {
  const raw = String(value ?? '').trim();
  if (!raw) return '';
  const upper = raw.toUpperCase();
  if (upper === 'SPRING' || raw === '春季') return 'SPRING';
  if (upper === 'SUMMER' || raw === '夏季') return 'SUMMER';
  if (upper === 'AUTUMN' || raw === '秋季') return 'AUTUMN';
  if (upper === 'WINTER' || raw === '冬季') return 'WINTER';
  if (upper === 'SPRING_SUMMER' || raw === '春夏') return 'SPRING_SUMMER';
  if (upper === 'AUTUMN_WINTER' || raw === '秋冬') return 'AUTUMN_WINTER';
  return raw;
};

File: backend/src/main/java/com/fashion/supplychain/style/util/StyleCategory.java
package com.fashion.supplychain.style.util;

import java.util.HashMap;
import java.util.Map;

/**
 * 品类规范化工具。
 *
 * <p>{@code t_style_info.category} 历史上混杂了中文标签、规范代码（WOMAN/SHIRT/JUPE 等）以及
 * 大小写英文（skirt/shirt/upper）等形态。前端表单/筛选统一用规范代码，因此这里把各种输入的
 * 品类值归一化为【规范代码】再落库或用于筛选；对没有规范代码的历史值（如"上衣""未分类"）保留原样。</p>
 */
public final class StyleCategory {

    private StyleCategory() {
    }

    /** raw（统一去空格并大写）→ 规范代码（或无语义代码时保留中文原样） */
    private static final Map<String, String> CODE = buildMap();

    private static Map<String, String> buildMap() {
        Map<String, String> m = new HashMap<>();
        // 主分类
        put(m, "WOMAN", "女装", "WOMEN", "LADY");
        put(m, "MAN", "男装", "MEN");
        put(m, "KIDS", "童装", "KID", "CHILD", "CHILDREN");
        put(m, "WCMAN", "女童装");
        put(m, "MCMAN", "男童装");
        put(m, "UNISEX", "男女同款");
        put(m, "SPORT", "运动装");
        put(m, "UNDERWEAR", "内衣");
        // 上装
        put(m, "T_SHIRT", "T恤", "TSHIRT", "T恤衫");
        put(m, "SHIRT", "衬衫", "BLOUSE");
        put(m, "HOODIE", "卫衣");
        put(m, "SWEATER", "毛衣", "KNITWEAR", "针织衫");
        put(m, "JACKET", "夹克");
        put(m, "COAT", "大衣");
        put(m, "TRENCH_COAT", "风衣");
        put(m, "DOWN_JACKET", "羽绒服");
        put(m, "PADDED_JACKET", "棉服");
        put(m, "SUIT", "西装");
        put(m, "VEST", "马甲");
        put(m, "BASE_SHIRT", "打底衫");
        // 下装
        put(m, "JUPE", "半身裙", "SKIRT", "裙子");
        put(m, "SHORTS", "短裤", "SHORT");
        put(m, "TROUSERS", "长裤", "PANTS", "裤子");
        put(m, "JEANS", "牛仔裤");
        put(m, "CASUAL_PANTS", "休闲裤");
        put(m, "SWEATPANTS", "运动裤");
        put(m, "BASE_PANTS", "打底裤", "LEGGINGS");
        // 连衣裙 / 连体
        put(m, "DRESS", "连衣裙", "ONEPIECE", "连体连衣裙");
        // 功能场景
        put(m, "YOGA_WEAR", "瑜伽服");
        put(m, "SUN_PROTECTION", "防晒服");
        put(m, "LOUNGEWEAR", "家居服");
        put(m, "SWIMWEAR", "泳装", "SWIMSUIT");
        put(m, "WORKWEAR", "工作服", "UNIFORM");
        // 无规范代码的历史值：统一保留中文标签
        put(m, "上衣", "上衣", "TOP", "TOPS", "UPPER", "上装");
        put(m, "未分类", "未分类", "UNDECLARED", "OTHER", "其它", "其他");
        return m;
    }

    private static void put(Map<String, String> m, String code, String chineseLabel, String... aliases) {
        m.put(code.toUpperCase(), code);
        m.put(chineseLabel.toUpperCase(), code);
        for (String a : aliases) {
            if (a != null) {
                m.put(a.toUpperCase().trim(), code);
            }
        }
    }

    /**
     * 把任意形态的品类值转成规范代码；空值原样返回（由上层决定是否填默认"未分类"）。
     */
    public static String normalize(String raw) {
        if (raw == null || raw.isEmpty()) {
            return raw;
        }
        String t = raw.trim();
        if (t.isEmpty()) {
            return raw;
        }
        String mapped = CODE.get(t.toUpperCase());
        return mapped != null ? mapped : t;
    }
}